server/paymentProviders/thegivingblock/index.js

```diff
@@ -7,27 +7,52 @@ import fetch from 'node-fetch';
 import orderStatus from '../../constants/order_status';
 import { TransactionTypes } from '../../constants/transactions';
 import { getFxRate } from '../../lib/currency';
+import logger from '../../lib/logger';
 import { getHostFee, getHostFeeSharePercent } from '../../lib/payments';
 import models from '../../models';
 
 const AES_ENCRYPTION_KEY = config.thegivingblock.aesEncryptionKey;
 const AES_ENCRYPTION_IV = config.thegivingblock.aesEncryptionIv;
 const AES_ENCRYPTION_METHOD = config.thegivingblock.aesEncryptionMethod;
 const API_URL = config.thegivingblock.apiUrl;
+const GIVINGBLOCK_USERNAME = config.thegivingblock.username;
+const GIVINGBLOCK_PASSWORD = config.thegivingblock.password;
+const MAX_REQUEST_RETRIES = 3;
 
 async function apiRequest(path, options = {}) {
   const response = await fetch(`${API_URL}${path}`, options);
   const result = await response.json();
 
-  if (result.data.errorMessage) {
-    throw new Error(`The Giving Block: ${result.data.errorMessage}`);
-  }
-
+  await handleErrors(result, path, options);
   // console.log(result);
 
   return result.data;
 }
 
+/*
+ * Whenever and api request is made we check if access token is expired and if so we login again.
+ * Refer: https://app.gitbook.com/@the-giving-block/s/public-api-documentation/#authentication-flow.
+ * Access tokens are only valid for 2 hours.
+ */
+async function handleErrors(result, path, options = {}) {
+  if (result.data.errorMessage) {
+    if (result.data.meta.errorCode === 'INVALID_JWT_TOKEN') {
+      logger.debug('Access token is invalid. Requesting a new one.');
+      let error;
+      for (let retries = 0; retries < MAX_REQUEST_RETRIES; retries++) {
+        try {
+          await login(GIVINGBLOCK_USERNAME, GIVINGBLOCK_PASSWORD);
+          await apiRequest(path, options);
```

Comment:
You're not refreshing the accessToken here ...

Reply:
Yep, my mistake. I've refactored the code.